### ISSUE TITLE
crash: Windows fatal exception: access violation

### ISSUE THREAD

**Issue by frspp** [jjenkner/ClusterPoints#6]
## Report Details

**Python Stack Trace**
```
Windows fatal exception: access violation

Thread 0x00004e14 (most recent call first):
  File "C:\Users/vy14ju/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\clusterpoints\ClusterPoints_algorithm.py", line 342 in processAlgorithm
    vlayer.commitChanges()

Current thread 0x00001560 (most recent call first):
  File "C:\OSGeo4W/apps/qgis/./python/plugins\processing\ProcessingPlugin.py", line 432 in executeAlgorithm
    dlg.exec_()
  File "C:\OSGeo4W/apps/qgis/./python/plugins\processing\gui\ProcessingToolbox.py", line 232 in executeAlgorithm
    self.executeWithGui.emit(alg.id(), self, self.in_place_mode, False)
```

**Stack Trace**

```

CPLODBCDriverInstaller::RemoveDriver :
QgsOgrProviderUtils::GDALOpenWrapper :
QgsMeshMemoryDataProvider::vertexCount :
QgsOgrProviderUtils::wildcards :
QgsMeshMemoryDataProvider::vertexCount :
QgsVectorLayerFeatureSource::QgsVectorLayerFeatureSource :
QgsVectorLayer::getFeatures :
QgsCachedFeatureWriterIterator::QgsCachedFeatureWriterIterator :
QgsVectorLayerCache::getFeatures :
QgsAttributeTableModel::loadLayer :
QObject::qt_static_metacall :
QMetaCallEvent::placeMetaCall :
QObject::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QCoreApplicationPrivate::sendPostedEvents :
qt_plugin_query_metadata :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QDialog::exec :
PyInit_QtWidgets :
PyArg_ParseTuple_SizeT :
PyEval_EvalFrameDefault :
PyObject_GC_Del :
PyFunction_Vectorcall :
PyFloat_FromDouble :
PyVectorcall_Call :
PyObject_Call :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
QObject::qt_static_metacall :
QMetaObject::activate :
PyInit_QtCore :
PyInit_QtCore :
PyType_GenericNew :
PyEval_EvalFrameDefault :
PyFunction_Vectorcall :
PyFloat_FromDouble :
PyVectorcall_Call :
PyObject_Call :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
QObject::qt_static_metacall :
QAbstractItemView::doubleClicked :
QTreeView::mouseDoubleClickEvent :
PyInit__gui :
QWidget::event :
QFrame::event :
QAbstractItemView::viewportEvent :
PyInit__gui :
QCoreApplicationPrivate::sendThroughObjectEventFilters :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :
```

**QGIS Info**
QGIS Version: 3.32.1-Lima
QGIS code revision: 4a078ecf
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.7.1
Running against GDAL: 3.7.1

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.22621

**Comment by jjenkner** [comment]
Thank you for the notice! The crash is related to some kind of output file type which my code has problems writing to. I'm considering changing the plugin code in a way that a new layer is created with the cluster information and the cluster information is no longer save to the original file. It seems too complicated to take care of all the different file types QGIS can write to ...

**Comment by jjenkner** [comment]
A new plugin version (6.1) has been released for QGIS 3.32. This version no longer writes the cluster output to the file system any more, but does create a new memory layer in QGIS instead. Consequently, any issues with uncommon file types are avoided from now on.
